server/src/test/java/org/elasticsearch/health/GetHealthResponseTests.java

```diff
@@ -0,0 +1,167 @@
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.is;
+
+public class GetHealthResponseTests extends ESTestCase {
+
+    public void testToXContent() throws IOException {
+        List<HealthIndicatorResult> indicatorResults = new ArrayList<>(2);
+        indicatorResults.add(createRandomIndicatorResult());
+        indicatorResults.add(createRandomIndicatorResult());
+        Response response = new Response(ClusterName.DEFAULT, indicatorResults, true);
+
+        XContentBuilder builder = XContentFactory.jsonBuilder().prettyPrint();
+        response.toXContentChunked().forEachRemaining(xcontent -> {
+            try {
+                xcontent.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            } catch (IOException e) {
+                logger.error(e.getMessage(), e);
+                fail(e.getMessage());
+            }
+        });
+        Map<String, Object> xContentMap = XContentHelper.convertToMap(BytesReference.bytes(builder), false, builder.contentType()).v2();
+        assertThat(xContentMap.get("status"), is(response.getStatus().xContentValue()));
+        assertXContentContainsExpectedResult(xContentMap, indicatorResults.get(0).name(), indicatorResults.get(0));
+        assertXContentContainsExpectedResult(xContentMap, indicatorResults.get(1).name(), indicatorResults.get(1));
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertXContentContainsExpectedResult(
+        Map<String, Object> xContentMap,
+        String indicatorName,
+        HealthIndicatorResult expectedIndicatorResult
+    ) {
+        Map<String, Object> allIndicators = (Map<String, Object>) xContentMap.get("indicators");
+        Map<String, Object> indicator = (Map<String, Object>) allIndicators.get(indicatorName);
+        List<Map<String, Object>> impacts = (List<Map<String, Object>>) indicator.get("impacts");
+        List<HealthIndicatorImpact> fromXContentImpacts = new ArrayList<>(impacts.size());
+        for (Map<String, Object> impact : impacts) {
+            List<ImpactArea> impactAreasList = new ArrayList<>();
+            List<String> fromXContentImpactAreas = (List<String>) impact.get("impact_areas");
+            for (String impactArea : fromXContentImpactAreas) {
+                impactAreasList.add(ImpactArea.valueOf(impactArea.toUpperCase(Locale.ROOT)));
+            }
+            String xcontentId = (String) impact.get("id");
+            fromXContentImpacts.add(
+                new HealthIndicatorImpact(
+                    indicatorName,
+                    // TODO add some methods to handle the id and xcontent id transitions
```

Comment:
will follow up in another PR as the whole stable/internal ID concept needs to be overhauled